modules/index-service/src/main/java/org/opencastproject/index/service/resources/list/provider/ContributorsListProvider.java

```diff
@@ -65,13 +70,30 @@
 
   private static final Logger logger = LoggerFactory.getLogger(ContributorsListProvider.class);
 
+  private Set<String> excludeUserProvider = new HashSet<>();
   private UserDirectoryService userDirectoryService;
   private AbstractSearchIndex searchIndex;
 
   protected void activate(BundleContext bundleContext) {
     logger.info("Contributors list provider activated!");
   }
 
+  public void updated(Dictionary<String, ?> properties) throws ConfigurationException {
```

Comment:
You should [use a declarative OSGi service with a `modified` method instead](https://docs.opencast.org/develop/developer/#proposal-log/#osgi-annotations-and-configuration). For a longer explanation take a look at the [webinar recording](https://recordings.rna1.blindsidenetworks.com/opencast/3cdef0a89e99c065d105b3a90fa2f5575d3cb628-1589986116901/capture/) or the [accompanying slides](https://docs.google.com/presentation/d/10iLzun3JcXfAEihOy5uxDKhyN0xtuYFb1dlBJcOPP4o/edit?usp=sharing).

Reply:
In general I'm agree with this suggestion. But then it will be the first and only one micro service in the index module with OSGI annotation.

Reply:
That's not what I meant. I get that introducing annotations here may be a problem (though we should if we can). I'm talking about using `modified`. That's something you can also specify in the XML code. Just set `modified="modified"` just like you would maybe set `activate="activate"`.

The problem with manually implemented managed services is that the configuration is loaded multiple times and in combination with a missing `modified`, the service will be restarted with any configuration change.

Reply:
Thanks, suggestion is now implemented.